core/src/main/java/io/micronaut/core/util/KotlinUtils.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.core.util;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.lang.reflect.Method;
+
+/**
+ * <p>Utility methods for working with Kotlin <code>suspend</code> functions</p>.
+ *
+ * @author Konrad Kamiński
+ * @since 1.3.0
+ */
+public class KotlinUtils {
+    /**
+     * Constant indicating whether coroutines are supported.
+     */
+    public static final boolean KOTLIN_COROUTINES_SUPPORTED;
+
+    static {
+        boolean areKotlinCoroutinesSupportedCandidate;
+
+        try {
+            Class.forName("kotlin.coroutines.Continuation"); // kotlin-stdlib
+            Class.forName("kotlin.reflect.jvm.ReflectJvmMapping"); // kotlin-reflect
```

Comment:
So I think we may need an associated PR to `micronaut-kotlin` to add the GraalVM metadata since these calls will fail in native image if not declared

Reply:
I think we actually should add it to `micronaut-core` since it's not obligatory to use `micronaut-kotlin` for Kotlin Micronaut apps.

Reply:
Yeah the issue with that is if Kotlin is not on the classpath then the GraalVM build will fail. So it cannot be in core 😞

Reply:
OK. I've found a solution which also works with GraalVM and doesn't need any metadata (see the latest changeset).